### ISSUE TITLE
[feature request] implement forward/stub-zone "allow-on" option

### ISSUE THREAD

**Issue by leoleovich** [NLnetLabs/unbound#300]
Sometimes it's necessary to branch logic of stub/forward zones. Real life example:
Unbound 1 - Listens on 1.2.3.4. Has access to internal authoritative server only.
Unbound 2 - Listens on 3.4.5.6. Has access to internal authoritative servers + internet recursion

Some services which don't need Internet recursion are using `1.2.3.4` and services which need Internet + internal recursion are using `3.4.5.6`.
These Unbound instances will be very similar. The main difference in the config file will be the interface they bind to. In addition the Internet-enabled instance will have the following section:
```
forward-zone:
    name: "."
    forward-addr: <recursive 1>
    forward-addr: <recursive 2>
```
We were thinking how we can optimise the setup and share resources (cache) between them. We suggesting the following option:
```
  forward-zone:
    name: "."
    forward-addr: <recursive 1>
    forward-addr: <recursive 2>
    allow-on: 3.4.5.6
```
In other words if query comes via `3.4.5.6` it will be forwarded via `forward-zone` to recursive. Otherwise - `REFUSED`.
Keep in mind the client may not be easily grouped by source IP. That's why we suggest the bind interface.
This would be valid other way around too! if we have an internal `stub-zone` which should only be accessible from `corpnet` let's say - we can define it like this:
```
  stub-zone:
    name: "mycorp."
    stub-addr: <internal authoritative DNS 1>
    stub-addr: <internal authoritative DNS 2>
    allow-on: 1.2.3.4
```

This has a potential to save up to 50% of the memory and give general flexibility.
Thank you.
